backend/src/routes/chat_route.py

```
@@ -0,0 +1,26 @@
+chatRouter=APIRouter()
+
+def get_db():
+    db=SessionLocal()
+    try:
+        yield db #iterator hai
+    finally:
+        db.close()
+
+
+@chatRouter.post("/chat")
+async def ChatResult(user_request:schema.ChatInput,db:Session=Depends(get_db),user:dict=Depends(getCurrentUser)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)